src/app/reporting/Engine.cpp

```diff
@@ -346,11 +435,6 @@ CHIP_ERROR Engine::SendReport(ReadHandler * apReadHandler, System::PacketBufferH
     mNumReportsInFlight++;
 
     err = apReadHandler->SendReportData(std::move(aPayload));
-
-    if (err != CHIP_NO_ERROR)
-    {
-        mNumReportsInFlight--;
```

Comment:
Can we move `mNumReportsInFlight++` to `apReadHandler->SendReportData`?

Reply:
mNumReportsInFlight is part of reporting engine